sympy/physics/quantum/tensorproduct.py

```diff
@@ -116,6 +117,17 @@ class TensorProduct(Expr):
     """
     is_commutative = False
 
+    @property
+    def is_Matrix(self):
+        return all(arg.is_Matrix for arg in self.args)
```

Comment:
Valid point. But I'm not sure how this implementation would work. For example, as a physicist, one would desire to use a `MatrixSymbol` of size `NxN` in which `N \to \infty`, which would mean that they are truly operators. But to have the freedom to interpolate between operators and "large `N`" matrices would really be desired for some branches of physics.

Separating "large `N`" matrices and operators to `KroneckerProduct` and `TensorProduct`, respectively,  would make this interpolation difficult (or?). This is my only worry. (is it the Issue #10771 you are thinking of?)

Then there is also the issue with `paulialgebra`, these are currently implemented as non-commutative `Symbol` objects, but are really matrices (most commonly represented by `2x2`matrices, and with the alternative implementation to PR #11598 (mentioned in the same PR) one would need to convert them to be `MatrixSymbol` objects instead.). These are often considered in tensor product combinations when it comes to, for example, Killing spinor equations.

My vision was to convert the current `paulialgebra` to `MatrixSymbol` objects so that these could be of more general use.